myldr/boot.c

```diff
@@ -204,6 +264,66 @@ typedef BOOL (WINAPI *pALLOW)(DWORD);
     /* extract embedded_files[0] (i.e. the custom Perl interpreter) 
      * into stmpdir (but under the same basename as argv[0]) */
     my_prog = par_findprog(argv[0], par_getenv("PATH"));
+
+#ifdef __MACH__
+    /* Detect if FAT binary */
+    FILE *obj_file = fopen(my_prog, "rb");
+    uint32_t magic = read_magic(obj_file, 0);
+    fclose(obj_file);
+    if (magic == FAT_CIGAM || magic == FAT_MAGIC) {
+
+      /* Create seperate dir for extracted thin binary*/
+      char *ftmpdir = malloc(strlen(stmpdir)+4);
+      ftmpdir = par_mktmpdir( argv );
+      strcat(ftmpdir, "_FAT");
+      sanitise_tmp(ftmpdir);
+      rc = my_mkdir(ftmpdir, 0700);
```

Comment:
I prefer to create a subdirectory, e.g. "thin" (w/o arch qualification) under `stempdir` and liposuct^H^H^H^Hextract `my_prog` into that with its original name (you can use `par_basename`). The reason is that boot.c also extracts
the custom perl interpreter into `stmpdir` as `par_basename(my_prog)` and finally execs that. So that when the user runs a packed executable `foo` they will see a `foo` in the process list.

Please use sprintf to create paths, it's hard to follow lotsa strcat's (also your first strcat into archthinbin below may not do what you expect, as malloc doesn't guarantee a zero-filled array).

Reply:
Fine, looking at it now.

Reply:
All done